### PR TITLE
remove the bugzilla checking for the openshift-tests-private repo

### PR DESCRIPTION
**Pull request by jianzhangbjz**
Hi dptp-helpdesk team,

I want to remove the `bugzilla/valid-bug` checking for the https://github.com/openshift/openshift-tests-private repo.
But, I couldn't find any related steps in the [ci-operator help doc](https://ci-operator-configresolver-ui-ci.apps.ci.l2s4.p1.openshiftapps.com/help). So, I update it based on my understanding.
Could you help have a review? Thanks very much!

### CODE REVIEW

**Review by petr-muller — CHANGES_REQUESTED** [review]
To disable requiring bugzilla-related labels for merges, you need to move the repo to the Tide query with approprriate criteria (see the current location https://github.com/openshift/release/blob/6b9c273940185c1df8fa57d082a668926b50ef9d/core-services/prow/02_config/_config.yaml#L1071 and how this query requires some labels to be present and some missing, and BZ-related ones are required in this query).

To disable the bugzilla bot from validating bugs in the title you can put `validate_by_default: false` in the bugzilla plugin config here: https://github.com/openshift/release/blob/6b9c273940185c1df8fa57d082a668926b50ef9d/core-services/prow/02_config/_plugins.yaml#L307

**Comment by jianzhangbjz** [comment]
@petr-muller Thanks for your help!
> To disable the bugzilla bot from validating bugs in the title you can put validate_by_default: false in the bugzilla plugin config here:

But, if I set the `validate_by_default: false`, will it affect other repos? I just want to disable the checking for this [openshift-tests-private](https://github.com/openshift/openshift-tests-private). Do we have a sample? Or docs? Thanks!

**Comment by petr-muller** [comment]
> @petr-muller Thanks for your help!
> 
> > To disable the bugzilla bot from validating bugs in the title you can put validate_by_default: false in the bugzilla plugin config here:
> 
> But, if I set the `validate_by_default: false`, will it affect other repos? I just want to disable the checking for this [openshift-tests-private](https://github.com/openshift/openshift-tests-private). Do we have a sample? Or docs? Thanks!

You can see in the existing config how a specific repo has even branch-specific settings. You can do the same on the repo level.

https://github.com/openshift/release/blob/6b9c273940185c1df8fa57d082a668926b50ef9d/core-services/prow/02_config/_plugins.yaml#L444-L449

**Comment by jianzhangbjz** [comment]
@petr-muller Many thanks! I modify it based on my understanding. But, I'm not sure where I still enabled the bugzilla plugin, could you help have a review? Thanks!
```console
{"component":"unset","file":"prow/cmd/checkconfig/main.go:78","func":"main.reportWarning","level":"warning","msg":"the following orgs or repos enable the bugzilla plugin but do not forbid the bugzilla/invalid-bug label for merging: [repo: openshift/openshift-tests repo: openshift/openshift-tests-private]","severity":"warning","time":"2020-09-29T10:19:39Z"}
```

**Comment by petr-muller** [comment]
@jianzhangbjz The BZ plugin is enabled org-wide and that will not change. That means that repositories need to forbid merging PRs with linked invalid bugs, just like advised by the bot. That's done in the configuration for `tide` (the merging bot) in the `_config.yaml` file. With checking disabled nothing will ever put the invalid bug label on your PR so it's essentially a no-op.

**Comment by jianzhangbjz** [comment]
@petr-muller Thanks for your explanation! So, how can I to `forbid the bugzilla/invalid-bug label for merging` and pass the test?

File: core-services/prow/02_config/_config.yaml [hunk 1]
```diff
@@ -710,6 +712,25 @@ tide:
...
+    - release-4.7
+    - master
+    labels:
+    - lgtm
+    - approved
+    missingLabels:
+    - needs-rebase
+    - do-not-merge/hold
+    - do-not-merge/work-in-progress
+    - do-not-merge/invalid-owners-file
```
Comment (petr-muller) [inline core-services/prow/02_config/_config.yaml:733]: This is close, but incorrect. For some reason you changed the configuration for the `ocs-operator` repos, which you shoudn't (note the differnt indent before the `includedBranches`.

You need to add a new list item (representing a GH query) that will set 1) labels that must be present (lgtm + approved) 2) labels that must be present (needs-rebase, the three do-not-merge one and the bugzilla/invalid-bug), the list of repos where this applies, and eiter the list of branches that should be included, or the list of branches taht should be excluded (implying that all others are inclded).

You should not modify any of the existing queries, just add a new one.

File: core-services/prow/02_config/_plugins.yaml [hunk 2]
```diff
@@ -8369,7 +8383,6 @@ plugins:
   - approve
   - assign
   - blunderbuss
-  - bugzilla
```
Comment (petr-muller) [inline core-services/prow/02_config/_plugins.yaml:8372]: Remove this. You must not disable plugins, the automation will enable them anyway.

File: core-services/prow/02_config/_plugins.yaml [hunk 3]
```diff
@@ -492,6 +492,20 @@ bugzilla:
...
+              - redhat
+              - qe_staff
+              validate_by_default: false
+        openshift-tests-private:
+          branches:
+            '*':
+              allowed_groups:
+              - redhat
+              - qe_staff
+              validate_by_default: false
```
Comment (petr-muller) [inline core-services/prow/02_config/_plugins.yaml:508]: This looks good.

File: core-services/prow/02_config/_plugins.yaml [hunk 4]
```diff
@@ -185,7 +185,12 @@ approve:
   repos:
   - openshift-priv/openshift-tests
   - openshift/openshift-tests
-  require_self_approval: false
+  require_self_approval: true
+- commandHelpLink: https://go.k8s.io/bot-commands
+  repos:
+  - openshift-priv/openshift-tests-private
+  - openshift/openshift-tests-private
```
Comment (petr-muller) [inline core-services/prow/02_config/_plugins.yaml:192]: This is basically the same settings lieke the `openshift-tests` repos above, so maybe just add these repos to the list of repos above?

File: core-services/prow/02_config/_plugins.yaml [hunk 5]
```diff
@@ -8394,7 +8407,6 @@ plugins:
   - approve
   - assign
   - blunderbuss
-  - bugzilla
```
Comment (petr-muller) [inline core-services/prow/02_config/_plugins.yaml:8397]: This too.

File: core-services/prow/02_config/_plugins.yaml (hunk 4 above)
Comment (jianzhangbjz) [reply core-services/prow/02_config/_plugins.yaml:192]: Yes, thanks!

File: core-services/prow/02_config/_plugins.yaml (hunk 2 above)
Comment (jianzhangbjz) [reply core-services/prow/02_config/_plugins.yaml:8372]: Ok, thanks!

File: core-services/prow/02_config/_config.yaml (hunk 1 above)
Comment (jianzhangbjz) [reply core-services/prow/02_config/_config.yaml:733]: Sorry for that, my mistake. Updated it.

**Review by petr-muller — APPROVED** [review]

**Pull request merged**